core/src/com/treode/notify/Notification.scala

```diff
@@ -23,14 +23,27 @@ package com.treode.notify
   * Inspired by [[http://martinfowler.com/articles/replaceThrowWithNotification.html Martin Fowler]].
   */
 
-class Notification {
```

Comment:
I'd recommend giving Notification two abstract methods: isEmpty and messages, and fill in the bodies for Errors and NoErrors. Actually, the body for messages is already filled in for Errors.